compiler/resolution/functionResolution.cpp

```diff
@@ -5596,6 +5597,7 @@ preFold(Expr* expr) {
       }
       call->replace(result);
     } else if (call->isPrimitive(PRIM_FIELD_ID_BY_NUM)) {
+      /* this is really for unions */
```

Comment:
It's only used for unions currently. It's not so clear to me why you would use it otherwise. But it *could* be called on say a record field.